examples/connext_dds/builtin_topics/cs/BuiltinTopicsExamplePublisher.cs

```diff
@@ -0,0 +1,173 @@
+
+            // Create the participant
+            participant = DomainParticipantFactory.Instance.CreateParticipant(
+                domainId,
+                qos: participantQos,
+                preEnableAction: p =>
+                {
+                    // The preEnableAction is executed right before the participant
+                    // is enabled and communication starts. By looking up the
+                    // built-in discovery readers here we ensure that
+                    // that they will receive all the discovery information.
+                    participantReader = p.BuiltinSubscriber
+                        .LookupDataReader<ParticipantBuiltinTopicData>(
+                            Subscriber.ParticipantBuiltinTopicName);
+
+                    // The DataAvailable event is called when another participant
+                    // is discovered and before this participant has started
+                    // communicating with it
+                    participantReader.DataAvailable += OnNewParticipant;
+
+                    subscriptionReader = p.BuiltinSubscriber
+                        .LookupDataReader<SubscriptionBuiltinTopicData>(
+                            Subscriber.SubscriptionBuiltinTopicName);
+
+                    subscriptionReader.DataAvailable += OnNewSubscription;
+                }
+            );
+
+            // Create a topic, a publisher and a writer
+            Topic<HelloWorld> topic = participant.CreateTopic<HelloWorld>(topicName);
+            Publisher publisher = participant.CreatePublisher();
+            writer = publisher.CreateDataWriter(topic);
+        }
+
+        /// <summary>
+        /// Publishes the data, which will only be received by the subscriptions
+        /// whose participants have set the right password
+        /// </summary>
+        public async Task Run(int sampleCount, CancellationToken cancellationToken)
+        {
+            var sample = new HelloWorld();
+            for (int count = 0;
+                count < sampleCount && !cancellationToken.IsCancellationRequested;
+                count++)
+            {
+                // Modify the data to be sent here
+                sample.x = (short)count;
+
+                Console.WriteLine($"Writing HelloWorld, count {count}");
+
+                writer.Write(sample);
+
+                await Task.Delay(1000, cancellationToken);
+            }
+        }
+
+        private void OnNewParticipant(AnyDataReader _)
+        {
+            // Take all the data about new participants
+            using LoanedSamples<ParticipantBuiltinTopicData> samples = participantReader
+                .Select()
+                .WithState(ViewState.New.And(InstanceState.Alive))
+                .Take();
+
+            foreach (var sample in samples)
+            {
+                if (!sample.Info.ValidData)
+                {
+                    continue;
```

Comment:
Maybe mention why this is necessary across this example code instead of ValidData()

Reply:
I can't find an explanation that is short and doesn't distract from the example. I'll leave it as is, since other examples do show the ValidData() enumerator.